internal/services/dynatrace/dynatrace_monitors_resource.go

```diff
@@ -0,0 +1,301 @@
+package dynatrace
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/identity"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/dynatrace/2023-04-27/monitors"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type MonitorsResource struct{}
+
+var _ sdk.ResourceWithUpdate = MonitorsResource{}
+
+type MonitorsResourceModel struct {
+	Name                          string                         `tfschema:"name"`
+	ResourceGroup                 string                         `tfschema:"resource_group_name"`
+	Location                      string                         `tfschema:"location"`
+	MonitoringStatus              bool                           `tfschema:"monitoring_enabled"`
+	MarketplaceSubscriptionStatus string                         `tfschema:"marketplace_subscription"`
+	Identity                      []identity.ModelSystemAssigned `tfschema:"identity"`
+	PlanData                      []PlanData                     `tfschema:"plan"`
+	UserInfo                      []UserInfo                     `tfschema:"user"`
+	Tags                          map[string]string              `tfschema:"tags"`
+}
+
+type PlanData struct {
+	BillingCycle  string `tfschema:"billing_cycle"`
+	EffectiveDate string `tfschema:"effective_date"`
+	PlanDetails   string `tfschema:"plan"`
+	UsageType     string `tfschema:"usage_type"`
+}
+
+type UserInfo struct {
+	Country      string `tfschema:"country"`
+	EmailAddress string `tfschema:"email"`
+	FirstName    string `tfschema:"first_name"`
+	LastName     string `tfschema:"last_name"`
+	PhoneNumber  string `tfschema:"phone_number"`
+}
+
+func (r MonitorsResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"monitoring_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			ForceNew: true,
```

Comment:
In most cases, this shall be enabled (as the resource is a monitor resource) when user wants to create it? Also as it is marked as `ForceNew: true`, does it make sense that we just remove this attribute and always set it as `true` when user wants to create it?

Otherwise, if this is togglable, then we may still keep this, but remove the `ForceNew: true`.

Reply:
This is not togglable for now. I will confirm with service team if it's ok to always set it as `true`.